src/main.rs

```diff
@@ -242,5 +279,93 @@ fn run() -> FeludaResult<()> {
     }
 
     log(LogLevel::Info, "Feluda completed successfully");
+
+    Ok(())
+}
+
+fn validate_ssh_key(key_path: &Path) -> Result<(), git2::Error> {
```

Comment:
Move both this and `clone_repository` functions to a separate file. Maybe `utils`